app/services/support_interface/organisation_permissions_export.rb

```diff
@@ -0,0 +1,78 @@
+module SupportInterface
+  class OrganisationPermissionsExport
+    def data_for_export
+      data = Audited::Audit
+        .joins('INNER JOIN provider_relationship_permissions prp ON audits.auditable_id = prp.id')
+        .joins('INNER JOIN providers tp ON prp.training_provider_id = tp.id')
+        .joins('INNER JOIN providers rp ON prp.ratifying_provider_id = rp.id')
+        .joins('LEFT JOIN provider_users_providers pup ON audits.user_id = pup.provider_user_id')
+        .joins('LEFT JOIN providers ups ON pup.provider_id = ups.id')
+        .where(auditable_type: 'ProviderRelationshipPermissions')
+        .where(action: %w[create update])
+        .where(permissions_clauses)
+        .pluck(
+          'audits.created_at',
+          'audits.user_id',
+          'audits.username',
```

Comment:
I think we don’t set `username` on the Audits. All my local ones are `username: nil` so this column in the CSV is empty for me. We may need to use the `user_id` and `user_type` to figure that out.

Reply:
I've attempted to fix this here https://github.com/DFE-Digital/apply-for-teacher-training/pull/4086/commits/4dbeb14f93693dfe0811959979afcaa7ab861df9 I've gone for the approach that doing as much db work as possible will help with performance, but this is undoubtedly resulting in a SQL-heavy export.

Reply:
yea, that is unfortunate :/